src/databricks/labs/lsql/dashboards.py

```diff
@@ -653,20 +657,39 @@ def update(self, other: "DashboardMetadata") -> None:
                 tile_metadatas.append(tile_metadata)
         self.tile_metadatas = tile_metadatas
 
-    def _get_tiles(
-        self, query_transformer: Callable[[sqlglot.Expression], sqlglot.Expression] | None = None
-    ) -> list[Tile]:
+    def replace_database(
+        self,
+        database: str,
+        *,
+        database_to_replace: str | None = None,
+    ) -> None:
+        """Replace the database in the queries.
+
+        Parameters :
+            database : str
+                The value to replace the database with
+            database_to_replace : str | None (default: None)
+                The database to replace, if None, all databases are replaced
+        """
+
+        def replace_database_in_query(node: sqlglot.Expression) -> sqlglot.Expression:
+            if (
+                isinstance(node, sqlglot.exp.Table)
+                and node.args.get("db") is not None
+                and (database_to_replace is None or getattr(node.args.get("db"), "this", "") == database_to_replace)
+            ):
+                node.args["db"].set("this", database)
+            return node
+
+        self.query_transformer = replace_database_in_query
```

Comment:
can you return an updated copy of `DashboardMetadata` instead of setting the `query_transformer` property?